corehq/apps/userreports/es/adapter.py

```diff
@@ -1,14 +1,83 @@
+import datetime
 from elasticsearch import NotFoundError
 from corehq.apps.userreports.util import get_table_name
 from corehq.apps.userreports.adapter import IndicatorAdapter
+from corehq.apps.es.es_query import HQESQuery
 from corehq.elastic import get_es_new
+from dimagi.utils.decorators.memoized import memoized
 from pillowtop.es_utils import INDEX_STANDARD_SETTINGS
 
 
 def normalize_id(values):
     return '-'.join(values).replace(' ', '_')
 
 
+class ESAlchemyResults(object):
+
+    def __init__(self, keys, values=dict):
+        self._keys = keys
+        if values:
+            for k in keys:
+                setattr(self, k, values[k])
```

Comment:
rather then setting attributes on this object you could implement `__getattr__` e.g. https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/api/resources/__init__.py#L19